qiskit/circuit/library/__init__.py

```diff
@@ -17,9 +17,45 @@
 
 .. currentmodule:: qiskit.circuit.library
 
-Standard Gates
+The circuit library is a collection of well studied circuits, directives, and gates that can be plugged
+into many experiments allowing users to program at higher levels of abstraction when developing and
+experimenting with circuits.
+
+
+Any element of the circuit library can be attached to a circuit with the method
+:meth:`.QuantumCircuit.append`. For example, to append a multi-controlled CNOT:
+
+.. jupyter-execute::
+
+    from qiskit.circuit.library import MCXGate
+    gate = MCXGate(4)
+
+    from qiskit import QuantumCircuit
+    circuit = QuantumCircuit(5)
+    circuit.append(gate, [0, 1, 4, 2, 3])
+    circuit.draw('text')
+
+The library is organized in several sections.
+
+Standard gates
 ==============
 
+These instructions are reversible unitary gates and they all subclass
+:class:`~qiskit.circuit.gate.Gate`. As a consequence, they all have the methods
+:meth:`~qiskit.circuit.gate.Gate.to_matrix`, :meth:`~qiskit.circuit.gate.Gate.power`,
+and :meth:`~qiskit.circuit.gate.Gate.control`.
+
+For example:
+
+.. jupyter-execute::
+
+    from qiskit.circuit.library import XGate
+    gate = XGate()
+    print(gate.to_matrix())             # X gate
+    print(gate.power(1/2).to_matrix())  # √X gate
+    print(gate.control(1).to_matrix())  # CX (controlled X) gate
```

Comment:
What does this actually output, isn't it just the repr for the gate object? Should we maybe put them in a circuit or something to show it more explicitly?

Reply:
it prints:
```
[[0.+0.j 1.+0.j]
 [1.+0.j 0.+0.j]]
[[0.5+0.5j 0.5-0.5j]
 [0.5-0.5j 0.5+0.5j]]
[[1.+0.j 0.+0.j 0.+0.j 0.+0.j]
 [0.+0.j 0.+0.j 0.+0.j 1.+0.j]
 [0.+0.j 0.+0.j 1.+0.j 0.+0.j]
 [0.+0.j 1.+0.j 0.+0.j 0.+0.j]]
```